BentoKit/BentoKit/Components/Button.swift

```diff
@@ -61,25 +61,67 @@ extension Component.Button {
             }
         }()
 
-        public let button = Button(type: .system).with {
-            $0.setContentHuggingPriority(.required, for: .vertical)
+        public var button = Button(type: .system)
+
+        fileprivate var buttonType: UIButton.ButtonType = .system {
+            didSet {
+                guard oldValue != buttonType else { return }
+
+                activityIndicator.removeFromSuperview()
```

Comment:
The logic here can be incorporated into `setupLayout()`, so we just need to invoke `setupLayout()` here to reconfigure as needed. Would centralize all these relevant bits too.